lib/presentation/molecules/pdf_viewer_molecule.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_cached_pdfview/flutter_cached_pdfview.dart';
+
+class PdfViewerMolecule extends StatelessWidget {
+  const PdfViewerMolecule(
+      {required this.url, super.key,
+      this.autoSpacing = true,
+      this.pageFling = true,
+      this.progressIndicatorColor = Colors.blue,});
+  final String url;
+  final bool autoSpacing;
+  final bool pageFling;
+  final Color progressIndicatorColor;
+
+  @override
+  Widget build(BuildContext context) {
+    return PDF(
+      autoSpacing: autoSpacing,
+      pageFling: pageFling,
+    ).cachedFromUrl(
+      url,
+      placeholder: (progress) => Center(
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+          children: [
+            CircularProgressIndicator(
```

Comment:
TimerAtom, Please fix

Reply:
That usage is just a loading indicator, I don't think Timer Atom would be appropriate for that as it is used to display timers. Let me know how you want to move forward.

Reply:
I would say to create a ProgressIndicatorAtom that use CircularProgressIndicator.

But here you know the progress, so you can pass it to our TimerAtom right?
![image](https://github.com/guyluz11/infinite_horizons/assets/9304740/cbd2c49d-d928-4c3a-a1b3-1b9e79344c68)

Reply:
My bad, I thought of ProgressIndicatorMolecule.
I even shared a screen shoot of it in the last comment.

Can you change it?

Reply:
Oh, that makes a lot more sense. I will make the required change.

Reply:
@guyluz11 I looked into this and since ProgressIndicatorMolecule uses an intrinsic controller, we can't use it in this context as we don't know in advance how long it will take to download the pdf. With TimerAtom I could control the controller as part of the progress function but I can't do the same with progress indicator. Do you want me to restructure progress indicator to allow for use with a parameterised controller as well?

Reply:
Yes please, restructure it so that we can control the percentage from outside